eclipse-theia/theia
# VSX: Add 'Install Another Version...' Command

#### What it does


Closes #9330, on which this PR is based.

- Supports the ability to install any compatible version of an user-installed extension, provided that the extension is available in the Open VSX Registry.

https://user-images.githubusercontent.com/46289281/114419822-c3067980-9b81-11eb-9155-6556d10b7919.mp4

> Original video from #9330 - the current iteration looks a bit different in practice.

#### How to test


1. Open the extensions view and locate the `Installed` view of all user-installed extensions.
2. Pick an extension --> Click the `manage` gear icon --> Click `Install Another Version...` command. Note that the command is disabled for any extension not available in the Open VSX Registry.
3. Select a compatible version to install from the Quick Pick dropdown.
4. Observe that the current version is marked with `Reload Required`, but the plugin continues working.
5. On reload, the newly installed version should be active.
6. If you're in a state displaying `Reload Required` and you run the command again and select the version currently running, `Reload Required` should change to `Uninstall`, and reloading should retain the same version.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: seantan22 <sean.a.tan@ericsson.com>
